flocker/node/agents/cinder.py

```diff
@@ -563,19 +563,35 @@ def detach_volume(self, blockdevice_id):
             transient_states=(u'in-use', u'detaching')
         )
 
-    def destroy_volume(self, blockdevice_id):
+    def destroy_volume(self, blockdevice_id, timeout=300):
+        """
+        Detach Cinder volume identified by blockdevice_id.
+
+        :raises TimeoutException: If the volume is not deleted
+        within the expected time
+        """
         try:
             self.cinder_volume_manager.delete(blockdevice_id)
         except CinderNotFound:
             raise UnknownVolume(blockdevice_id)
-
-        while True:
-            # Don't loop forever here.  FLOC-1853
+        start_time = time.time()
+        #NOTE: maybe we could share the ''VOLUME_STATE_CHANGE_TIMEOUT''
+        #in the ebs implementation
+        #TODO do we need to check the state instead of waiting until the volume
```

Comment:
That would be another legitimate alternative. I think there is a "deleting" state, and it seems reasonable to just wait until that shows up. There is however some expectation that deleted volumes don't show up in list_volume(), so we may need to wait until it's actually not there. So this is fine.